src/corehost/corehost.cpp

```diff
@@ -151,6 +146,29 @@ int run(const int argc, const pal::char_t* argv[])
         trace::error(_X("A fatal error was encountered. This executable was not bound to load a managed DLL."));
         return StatusCode::AppHostExeNotBoundFailure;
     }
+#else
+    pal::string_t own_name = get_executable(get_filename(own_path));
+
+    if (pal::strcasecmp(own_name.c_str(), _X("dotnet")) != 0)
+    {
+        trace::error(_X("Fatal error : dotnet executable has been renamed to %s."), own_name.c_str());
+        return StatusCode::CoreHostEntryPointFailure;
```

Comment:
Looks good. I would suggest that error message be more actionable - "Cannot execute dotnet when renamed to ...."